berkeley-mobile/StudyPact/Profile/ProfileViewController.swift

```diff
@@ -60,35 +58,18 @@ class ProfileViewController: UIViewController, UITextFieldDelegate, UINavigation
             loggedOutView()
             return
         }
-        StudyPact.shared.registerUser(user: user) { success in
-            if !success {
-                GIDSignIn.sharedInstance().signOut()
-                DispatchQueue.main.async {
-                    self.loggedOutView()
-                }
-                return
-            }
-            self.loggedInView()
-            let fullName = user.profile.name
-            let email = user.profile.email
-            StudyPact.shared.email = email
-            
-            self.initialsLabel.text = String(fullName?.prefix(1) ?? "")
-            self.fullNameField.text = fullName
-            self.emailTextField.textField.text = email
-            
-            if user.profile.hasImage {
-                guard let imageUrl = user.profile.imageURL(withDimension: 250) else { return }
-                ImageLoader.shared.getImage(url: imageUrl) { result in
-                    switch result {
-                    case .success(let image):
-                        DispatchQueue.main.async() { [weak self] in
-                            self!.profileImage = image
-                        }
-                    case .failure(let error):
-                        print(error)
+        // if new sign in to google rather than previous sign in
+        if StudyPact.shared.getCryptoHash() == nil {
+            StudyPact.shared.registerUser(user: user) { success in
```

Comment:
What do you think about moving this block to `SignInManager` before this method is called on the delegates, so all the authentication is centralized there?

Reply:
so essentially this is if the user wasn't previously signed in and has just signed in using the "Sign in with Google" button. so this should theoretically only happen on the Profile page. the check that the cryptohash is nil is essentially checking for this.
we want this because we only want to register the user if the cryptohash expired/doesn't exist and the user just signed into google